lib/datasets/penguins.rb

```diff
@@ -112,14 +111,37 @@ def each(&block)
     end
 
     private def convert_record(raw_record)
-      Record.new(raw_record.species.split(' ')[0],
-                 raw_record.island,
-                 raw_record.culmen_length_mm,
-                 raw_record.culmen_depth_mm,
-                 raw_record.flipper_length_mm&.to_i,
-                 raw_record.body_mass_g&.to_i,
-                 raw_record.sex&.downcase,
-                 raw_record.date_egg&.year)
+      sex = raw_record.sex&.downcase
+      Record.new(*cleansing!(raw_record))
+    end
+
+    private def cleansing!(raw_record)
```

Comment:
Can we remove `!`? Because this doesn't change `raw_record` destructively.

`cleanse` may be better than `cleansing`.

Reply:
I'll rename it to `cleanse_fields`